### PR TITLE
GS: Cleanup GSState.cpp.

### PR DESCRIPTION
**Pull request by lightningterror**
### Description of Changes

GS: Cleanup GSState.cpp.
constants, scope, duplicate declarations, initializations.
### Rationale behind Changes

Cleaner code.
### Suggested Testing Steps

Still compiles, nothing broke.

### CODE REVIEW

File: pcsx2/GS/GSState.cpp [hunk 1]
```diff
@@ -3111,34 +3111,34 @@ __forceinline void GSState::VertexKick(u32 skip)
 
 	// callers should write XYZUVF to m_v.m[1] in one piece to have this load store-forwarded, either by the cpu or the compiler when this function is inlined
 
-	GSVector4i v0(m_v.m[0]);
-	GSVector4i v1(m_v.m[1]);
+	GSVector4i tail_v0(m_v.m[0]);
+	GSVector4i tail_v1(m_v.m[1]);
```
Comment (refractionpcsx2) [inline pcsx2/GS/GSState.cpp:3115]: they aren't the tail, they're going in to the the tail (further down), they are the new vertex information

Comment (lightningterror) [reply pcsx2/GS/GSState.cpp:3115]: `new_v0/1` then, wanted to avoid having them as duplicates with the ones a bit down.

Comment (refractionpcsx2) [reply pcsx2/GS/GSState.cpp:3115]: sure

File: pcsx2/GS/GSState.cpp [hunk 2]
```diff
@@ -3111,34 +3111,34 @@ __forceinline void GSState::VertexKick(u32 skip)
...
 
 	GSVector4i* RESTRICT tailptr = (GSVector4i*)&m_vertex.buff[tail];
 
-	tailptr[0] = v0;
-	tailptr[1] = v1;
+	tailptr[0] = new_v0;
+	tailptr[1] = new_v1;
 
-	const GSVector4i xy = v1.xxxx().u16to32().sub32(m_ofxy);
+	const GSVector4i xy = new_v1.xxxx().u16to32().sub32(m_ofxy);
```
Comment (tellowkrinkle) [inline pcsx2/GS/GSState.cpp:3122]: Why the new name?

Comment (refractionpcsx2) [reply pcsx2/GS/GSState.cpp:3122]: It was to prevent confusion with them being reused further down.

**Pull request merged**